docs/pt/docs/advanced/websockets.md

```diff
@@ -0,0 +1,257 @@
+# WebSockets
+
+Você pode usar  <a href="https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API" class="external-link" target="_blank">WebSockets</a> com **FastAPI**.
+
+## Instalando `WebSockets`
+
+Garanta que você criou um [ambiente virtual](../virtual-environments.md){.internal-link target=_blank}, o ativou e instalou o `websockets`:
+
+<div class="termy">
+
+```console
+$ pip install websockets
+
+---> 100%
+```
+
+</div>
+
+## Cliente WebSockets
+
+### Em produção
+
+Em seu sistema de produção, você provavelmente tem um frontend criado com um framework moderno como React, Vue.js ou Angular.
+
+E para comunicar usando WebSockets com seu backend, você provavelmente usaria as utilidades do seu frontend.
+
+Ou você pode ter um aplicativo móvel nativo que se comunica diretamente com seu backend WebSocket, em código nativo.
+
+Ou você pode ter qualquer outra forma de comunicar com o endpoint WebSocket.
+
+---
+
+Mas para este exemplo, usaremos um documento HTML muito simples com algum JavaScript, tudo dentro de uma string longa.
+
+Esse, é claro, não é o ideal e você não o usaria para produção.
+
+Na produção, você teria uma das opções acima.
+
+Mas é a maneira mais simples de focar no lado do servidor de WebSockets e ter um exemplo funcional:
+
+```Python hl_lines="2  6-38  41-43"
+{!../../docs_src/websockets/tutorial001.py!}
+```
+
+## Criando um `websocket`
+
+Em sua aplicação **FastAPI**, crie um `websocket`:
+
+```Python hl_lines="1  46-47"
+{!../../docs_src/websockets/tutorial001.py!}
+```
+
+/// note | "Detalhes Técnicos"
+
+Você também poderia usar `from starlette.websockets import WebSocket`.
+
+A **FastAPI** fornece o mesmo `WebSocket` diretamente apenas como uma conveniência para você, o desenvolvedor. Mas ele vem diretamente do Starlette.
+
+///
+
+## Aguardar por mensagens e enviar mensagens
+
+Em seu rota WebSocket você pode `await` por mensagens e enviar mensagens.
+
+```Python hl_lines="48-52"
+{!../../docs_src/websockets/tutorial001.py!}
+```
+
+Você pode receber e enviar dados binários, de texto e JSON.
+
+## Tente você mesmo
+
+Se seu arquivo for nomeado `main.py`, execute sua aplicação com:
+
+<div class="termy">
+
+```console
+$ fastapi dev main.py
+
+<span style="color: green;">INFO</span>:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
+```
+
+</div>
+
+Abra seu broser em: <a href="http://127.0.0.1:8000" class="external-link" target="_blank">http://127.0.0.1:8000</a>.
```

Comment:
missed the **w** in browser: Abra seu bro**w**ser em...

But I think you can use:

Abra o seu navegador em...